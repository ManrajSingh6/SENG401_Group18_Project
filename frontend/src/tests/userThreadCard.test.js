import React from 'react';
import renderer from 'react-test-renderer';
import ThreadCard from '../components/userThreadCard';
import '@testing-library/jest-dom';
import { render, screen, fireEvent,cleanup} from '@testing-library/react';
import ThumbUpRoundedIcon from '@mui/icons-material/ThumbUpRounded';
import ArticleIcon from '@mui/icons-material/Article';


// testing delete button
jest.mock('../components/userThreadCard', () => {
    const threadCard = () => (
      <div data-testid='threadCard'>user thread card</div>
    );
  
    return threadCard;
  });
  
  
it('Confirm deletion button', ()=>{
render(<ThreadCard/>);
const button = screen.getByTestId('threadCard', {name: "button"});
fireEvent.click(button);
expect(screen.getByTestId("threadCard")).toBeInTheDocument();
});

//testing like icon
test('renders ThumbUpRoundedIcon', () => {
    const testme = renderer.create(<ThumbUpRoundedIcon />).toJSON();
    expect(testme).toMatchSnapshot();
  });
  
  // testing comment icon
  test('renders ArticleIcon', () => {
      const testme = renderer.create(<ArticleIcon />).toJSON();
      expect(testme).toMatchSnapshot();
    });