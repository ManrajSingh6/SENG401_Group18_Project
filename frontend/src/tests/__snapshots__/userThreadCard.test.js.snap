// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArticleIcon 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ArticleIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-5 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
  />
</svg>
`;

exports[`renders ThumbUpRoundedIcon 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ThumbUpRoundedIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M13.12 2.06 7.58 7.6c-.37.37-.58.88-.58 1.41V19c0 1.1.9 2 2 2h9c.8 0 1.52-.48 1.84-1.21l3.26-7.61C23.94 10.2 22.49 8 20.34 8h-5.65l.95-4.58c.1-.5-.05-1.01-.41-1.37-.59-.58-1.53-.58-2.11.01zM3 21c1.1 0 2-.9 2-2v-8c0-1.1-.9-2-2-2s-2 .9-2 2v8c0 1.1.9 2 2 2z"
  />
</svg>
`;
